### PR TITLE
LPS-128451 Escape content to avoid running embedded Js code

### PR DESCRIPTION
**Pull request by boton**
**Steps to reprodue:**
1. Go Content & Data > Blogs > Add a new blog entry
1. Fill up the title or subtitle with `">`

**Result:** 
An alert pops up

**Expected:** 
Not parse HTML

Previously revised liferay-lima#1561

### CODE REVIEW

**Review by wincent — APPROVED** [review]
Looks reasonable to me. With fixes like this, especially seeing as this [doesn't repro broadly across other branches](https://issues.liferay.com/browse/LSV-809), I like to understand when/how this problem arose and to see whether we could have caught it in review. I see the code came in via [this commit](https://github.com/brianchandotcom/liferay-portal/pull/87446/commits/e6e674be0789a7a99e553aa2454163ca7eed0b23), but it's hard to follow the review trail if you look at [the PR](https://github.com/austinchiang/liferay-portal/pull/752) that was forwarded to Brian.

Searching in my email pulls up these PRs, but none of them contain the bad commit (as far as I can see):

- https://github.com/ambrinchaudhary/liferay-portal/pull/305
- https://github.com/boton/liferay-portal/pull/150
- https://github.com/julien/liferay-portal/pull/238

Last resort: looking at [PRs](https://github.com/julien/liferay-portal/pulls?q=is%3Apr+LPS-110007) in @julien's fork, only [this one](https://github.com/julien/liferay-portal/pull/239) has the bad commit in it, which is the first visible trace that I can find of it in all this. It then went to @ambrinchaudhary's fork [here](https://github.com/ambrinchaudhary/liferay-portal/pull/320), and from there to @austinchiang's [here](https://github.com/austinchiang/liferay-portal/pull/737) → finally after all of that it got resent one more time and from there we see it making its way to Brian. Mystery "solved", but probably not in the most satisfactory way, because in all of those PRs there wasn't really any discussion of this part of the code. I don't know if anybody has any suggestions beyond "look carefully any time we set `innerHTML`" (we do have [this issue](https://github.com/liferay/liferay-frontend-projects/issues/345) to create a security document/checklist, but haven't started on it yet, and we may be able to identify areas for deeper auditing based on simple patterns that we know to be risky). 🤷

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by jbalsas** [comment]
@ambrinchaudhary, don't be shy with that `ci:forward` command! 😉 

As per @wincent's comment, there's always been talks about having a more controlled "printing" environment that made it easier for us to detect cases like this. @csierra mentioned their team was working on something not too long ago but might not even apply in this scenario, so I think best we can do for now is start identifying the patterns and lint for them when possible and/or document them otherwise.

**Comment by jbalsas** [comment]
ci:forward

**Comment by csierra** [comment]
hey @jbalsas, 
yes, what we were working on would not apply for this scenario. 
Alas, no decision has been made yet. 

Carlos.
